File: 力扣习题/面试题 05.02二进制数转字符串/problems.ts
function printBin(num: number): string {
    const res: string[] = ['0', '.'];
    let k = 0.5
    while (res.length < 32) {
        if (num - k > 0) {
            res.push('1');
            num -= k;
            k *= 0.5;
        } else if(num - k == 0) {
            num -= k;
            res.push('1');
            break;
        } else {
            res.push('0');
            k *= 0.5;
        }
    }
    if (num != 0) {
        return 'ERROR';
    } 
    return res.join('');

};

File: 力扣习题/面试题 05.02二进制数转字符串/problems.py
class Solution:
    def printBin(self, num: float) -> str:
        res = ['0', '.']
        k = 0.5
        while len(res) < 32:
            if num - k > 0:
                num -= k
                k *= 0.5
                res.append('1')
            elif num - k == 0:
                num -= k
                res.append('1')
                break
            else:
                res.append('0')
                k *= 0.5
        return 'ERROR' if num != 0 else ''.join(res)